File: euler_3/euler_3.c
#include <stdio.h>
#include <math.h>

int main()
{
    long num = 600851475143;
    int test = 2;

    while (pow(test, 2) < num) {
        if (num % test == 0)
            num = num / test;
        test += 1;
    }

    printf("The answer is %d\n", num);

    return 0;
}

File: euler_3/euler_3.clj
(def number 600851475143)

(defn pow [base exp]
  (reduce * (repeat exp base)))

(time (loop [high number
        test 2]
   (if (< high (pow test 2))
     (println "The answer is:" high)
     (let [next-test (inc test)
           next-high (if (zero? (mod high test)) (/ high test) high)]
       (recur next-high next-test)
       ))))
